guides/source/active_model_basics.md

```diff
@@ -285,101 +804,308 @@ irb> person = Person.new
 irb> person.token = "2b1f325"
 irb> person.valid?
 => false
-irb> person.name = 'vishnu'
-irb> person.email = 'me'
+
+irb> person.name = "vishnu"
+irb> person.email = "me"
 irb> person.valid?
 => false
-irb> person.email = 'me@vishnuatrai.com'
+
+irb> person.email = "me@vishnuatrai.com"
 irb> person.valid?
 => true
+
+# token uses validate! and will raise an exception when not set.
 irb> person.token = nil
 irb> person.valid?
-ActiveModel::StrictValidationFailed
+=> "Token can't be blank (ActiveModel::StrictValidationFailed)"
+```
+
+#### Validation Methods and Options
+
+You can add validations using some of the following methods:
+
+- `validate`: Adds validation through a method or a block to the class. You can
+  read more about how to use `validate`
+  [here](https://api.rubyonrails.org/classes/ActiveModel/Validations/ClassMethods.html#method-i-validate).
```

Comment:
I can add more details about how to use `validate` here, it could be useful, it was duplicating whats in the API guide and I'd given a base example already so instead I referenced the link.   

Throughout the guide, I've tried to balance between outlining the details in the guides vs linking to the outlined details in the API. I made a lot of judgment calls based on the information being conveyed. I hope that I got that balance right, but I'm open to feedback if anyone thinks I could be detailing more or less.